src/main/java/vivatech/init/VivatechItems.java

```diff
@@ -22,10 +27,14 @@
         ENERGY_CONDUIT = new BlockItem(VivatechBlocks.ENERGY_CONDUIT, Vivatech.ITEM_SETTINGS);
         COAL_GENERATOR = new BlockItem(VivatechBlocks.COAL_GENERATOR, Vivatech.ITEM_SETTINGS);
         CRUSHER = new BlockItem(VivatechBlocks.CRUSHER, Vivatech.ITEM_SETTINGS);
-        ELECTRIC_FURNACE = new BlockItem(VivatechBlocks.ELECTRIC_FURNACE, Vivatech.ITEM_SETTINGS);
         ENERGY_BANK = new BlockItem(VivatechBlocks.ENERGY_BANK, Vivatech.ITEM_SETTINGS);
         PRESS = new BlockItem(VivatechBlocks.PRESS, Vivatech.ITEM_SETTINGS);
         BATTERY = new BatteryItem();
+        
+        ELECTRIC_FURNACE = new Item[MachineTiers.values().length];
+        ELECTRIC_FURNACE[MachineTiers.MINIMAL.ordinal()] = new BlockItem(VivatechBlocks.ELECTRIC_FURNACE[MachineTiers.MINIMAL.ordinal()], Vivatech.ITEM_SETTINGS);
+        ELECTRIC_FURNACE[MachineTiers.NORMAL.ordinal()] = new BlockItem(VivatechBlocks.ELECTRIC_FURNACE[MachineTiers.NORMAL.ordinal()], Vivatech.ITEM_SETTINGS);
+        ELECTRIC_FURNACE[MachineTiers.ADVANCED.ordinal()] = new BlockItem(VivatechBlocks.ELECTRIC_FURNACE[MachineTiers.ADVANCED.ordinal()], Vivatech.ITEM_SETTINGS);
```

Comment:
From L33 to here is a bit mess. Same about block registration.

Reply:
Yeah, you're right. How would you prefer I cut down the mess? With a helper method or something else?

Reply:
Helper. Try to give it a meaningful name and don’t do side-effects (like registering an item inside block registry helper)